subsys/bluetooth/controller/ll_sw/nordic/lll/lll_peripheral_iso.c

```diff
@@ -1305,3 +1263,147 @@ static void isr_done(void *param)
 
 	lll_isr_cleanup(param);
 }
+
+static void payload_count_txrx_flush_or_tx_inc(struct lll_conn_iso_stream *cis_lll)
+{
+	if (cis_lll->tx.bn) {
+		uint64_t payload_count;
+		uint8_t u;
+
+		payload_count = cis_lll->tx.payload_count + cis_lll->tx.bn_curr - 1U;
+		u = cis_lll->nse - ((cis_lll->nse / cis_lll->tx.bn) *
+				    (cis_lll->tx.bn - 1U -
+				     (payload_count % cis_lll->tx.bn)));
+		if ((((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr +
+		      cis_lll->tx.ft + u) <
+		     (cis_lll->event_count + se_curr + 2U)) ||
+		    ((((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr) >
+		      cis_lll->event_count) &&
+		     ((cis_lll->tx.payload_count / cis_lll->tx.bn) < cis_lll->event_count))) {
+			uint8_t sn_update = cis_lll->tx.bn + 1U - cis_lll->tx.bn_curr;
+
+			/* sn and nesn are 1-bit, only Least Significant bit is needed */
+			cis_lll->sn += sn_update;
+			cis_lll->tx.payload_count += cis_lll->tx.bn;
+			cis_lll->tx.bn_curr = 1U;
+		}
+	}
+
+	if (!has_tx) {
+		has_tx = 1U;
+		se_curr = 1U;
+	}
+
+	if (cis_lll->rx.bn) {
```

Comment:
Ditto

Reply:
Regarding refactoring, it will be addressed in #63152.